File: app/src/main/java/in/nikhil/cinestine/Adapters/ReviewAdapter.java
package in.nikhil.cinestine.Adapters;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;

import in.nikhil.cinestine.Model.Review;
import in.nikhil.cinestine.Model.Trailer;
import in.nikhil.cinestine.R;

/**
 * Created by nikhil on 28-08-2016.
 */
public class ReviewAdapter extends RecyclerView.Adapter<ReviewAdapter.MyViewHolder> {

  private ArrayList<Review> ReviewList = new ArrayList<Review>();
  private LayoutInflater inflater;
  private Context context;

  public ReviewAdapter(Context context) {
    inflater = LayoutInflater.from(context);
    this.context = context;
  }

  public void setReviewsList(ArrayList<Review> reviewList) {
    ReviewList = reviewList;
    notifyDataSetChanged();
  }

  @NonNull
  @Override
  public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
    View v = inflater.inflate(R.layout.recycler_review_single_row, parent, false);
    return new MyViewHolder(v);
  }

  @Override
  public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
    final Review current = ReviewList.get(position);
    String author = context.getString(R.string.author) + current.getAuthor();
    holder.author.setText(author);
    holder.content.setText(current.getContent());
  }

  @Override
  public int getItemCount() {
    return ReviewList.size();
  }

  class MyViewHolder extends RecyclerView.ViewHolder {

    TextView author;
    TextView content;

    MyViewHolder(View itemView) {
      super(itemView);
      author = itemView.findViewById(R.id.review_single_row_author);
      content = itemView.findViewById(R.id.review_single_row_content);
    }
  }
}

File: app/src/main/java/nikhil/cinestine/ui/details/ReviewAdapter.kt
package nikhil.cinestine.ui.details

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import nikhil.cinestine.R
import nikhil.cinestine.databinding.RecyclerReviewSingleRowBinding
import nikhil.cinestine.model.Review

class ReviewAdapter : ListAdapter<Review, ReviewAdapter.ReviewViewHolder>(Diff) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ReviewViewHolder {
        val binding = RecyclerReviewSingleRowBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ReviewViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ReviewViewHolder, position: Int) {
        holder.bind(getItem(position))
    }

    class ReviewViewHolder(
        private val binding: RecyclerReviewSingleRowBinding
    ) : RecyclerView.ViewHolder(binding.root) {
        fun bind(review: Review) {
            val author = review.author.ifBlank { "?" }
            binding.reviewSingleRowAuthor.text = binding.root.context.getString(R.string.author_format, author)
            binding.reviewAuthorInitial.text = author.first().uppercaseChar().toString()
            binding.reviewSingleRowContent.text = review.content
        }
    }

    private companion object {
        val Diff = object : DiffUtil.ItemCallback<Review>() {
            override fun areItemsTheSame(oldItem: Review, newItem: Review) = oldItem.id == newItem.id
            override fun areContentsTheSame(oldItem: Review, newItem: Review) = oldItem == newItem
        }
    }
}
